src/patterns/filtering/multiple-categories.stories.ts

```diff
@@ -0,0 +1,200 @@
+                [hideLabel]="hideLabel"
+                [checked]="listItem.checked"
+                [value]="listItem.value"
+                (change)="onCheckboxChange($event)">
+                {{ listItem.value }}
+            </ibm-checkbox>
+        </fieldset>
+    </div>
+    <div style="float: left;">
+        <ibm-table-container>
+        <ibm-table
+            style="display: block; width: 650px;"
+            [model]="model"
+            size="lg"
+            [showSelectionColumn]="false">
+            <ng-content></ng-content>
+        </ibm-table>
+        </ibm-table-container>
+    </div>
+    `
+})
+class SampleMultiCategories{
+    model = new TableModel();
+    checkBoxFilters = new Set();
+    radioFilter = null;
+
+    dataset = [
+        { name: "Apple", type: ["Fruit"], color: "Red" },
+        { name: "Grape", type: ["Fruit"], color: "Purple" },
+        { name: "Eggplant", type: ["Fruit"], color: "Purple" },
+        { name: "FruitVegMeat", type: ["Fruit", "Vegetable", "Meat"], color: "White" },
+        { name: "Lettuce", type: ["Vegetable"], color: "Green" },
+        { name: "Daikon Radish", type: ["Vegetable"], color: "White" },
+        { name: "Beef", type: ["Meat"], color: "Red" }
+    ];
+
+    manyRadios = [
+        { color: "Red", checked: false },
+        { color: "Purple", checked: false },
+        { color: "Green", checked: false },
+        { color: "White", checked: false }
+    ]
+
+    checkboxList = [
+        { value: "Fruit", checked: false},
+        { value: "Vegetable", checked: false},
+        { value: "Meat", checked: false}
+    ]
+
+    onCheckboxChange(event) {
+        if (event.checked) {
+            this.checkBoxFilters.add(event.source.value);
+        } else {
+            this.checkBoxFilters.delete(event.source.value);
+        }
+        this.applyFilters();
+    }
+
+    onRadioChange(event) {
+        this.radioFilter = event.value;
+        this.applyFilters();
+    }
+
+    resetFilters() {
+        this.manyRadios = this.manyRadios.map(obj => {
+            return { color: obj.color, checked: false }
+        });
+
+        this.checkboxList = this.checkboxList.map(obj => {
+            return { value: obj.value, checked: false }
+        });
+
+        this.radioFilter = null;
+        this.checkBoxFilters = new Set();
+
+        this.applyFilters();
+    }
+
+    applyFilters() {
+        this.model.data = [];
+        this.model.data.pop();
+
+        this.dataset.forEach(data => {
+            if((this.applyCheckboxFilters(data.type) || !this.checkBoxFilters.size) && (data.color === this.radioFilter || !this.radioFilter) ) {
+                this.model.data.push(
+                    [
+                        new TableItem({ data: data.name }),
+                        new TableItem({ data: "Lorem ipsum dolor sit amet, consectetur adipiscing elit." })
+                    ]
+                )
+            }
+        });
```

Comment:
I'm not sure what a lot of this is doing ... could this be simplified with a `this.model.data = this.dataset.filter(...)`?